Packaging fixes for Debian, clean deprecated buster from tests

wagner-intevation opened the pull request:
Is already in home:sebix:intelmq:unstable and the Debian 12 build works.

wagner-intevation committed:
doc: remove misplaced changelog entry

wagner-intevation committed:
pkg: control: fix maintainer and dependencies

most dependencies lacked the minimum required version
that lead to probably false assumptions about installable packages on
too old distros

wagner-intevation committed:
pkg: github_api remove obsolete requirements

requests is a core dependency, we do not need to require it for the bot

aaronkaplan commented on debian/control:
```diff
@@ -37,7 +37,7 @@ Depends: bash-completion,
          python3-psutil (>= 1.2.1),
          python3-redis (>= 2.10),
-         python3-requests (>= 2.2.1),
+         python3-requests (>= 2.2.0),
```
why the downgrade?

sebix replied:
to align it with `setup.py`

aaronkaplan commented on CHANGELOG.md:
```diff
@@ -95,12 +95,6 @@ This is short list of the most important known issues. The full list can be retr
 - Corrupt dump files when interrupted during writing (#870).
 
 
-3.0.2 (2021-09-10)
```
does it make sense to remove this?

sebix replied:
It's a duplicate/leftover. The correct 3.0.2 entry is much longer.

sebix committed:
pkg: rules: use default dh_auto_install for install step

the previous install was hardcoded and outdated on Debian 12
use the distro-default, fixes the build

fixes certtools/intelmq#2384

sebix committed:
pkg: add 3.2.2-1 changelog snippet for debian

wagner-intevation commented:
Rebased the commits on current develop and added an intermediate revision 3.2.1-2, which we could release straightaway

sebix committed:
remove Debian 10 buster from tests and documentation

Support for Debian 10 Buster was dropped in PR#2352
https://github.com/certtools/intelmq/pull/2352

wagner-intevation commented:
Replacing buster by bullseye in the intelmq tests also required this change in the vagrant test code: https://github.com/certtools/intelmq-vagrant/commit/f824c7923e6b70b3e2ea0a4bb19f1656135a898d

wagner-intevation commented:
Some cleanup work after https://github.com/certtools/intelmq/pull/2352 was missing (buster is no longer supported) and that failed the tests here. Now everything succeeds.

kamil-certat reviewed (approved):
LGTM, would you consider fixing packaging tests in `intelmq-api`? Would be nice to have them working

sebix commented:
> would you consider fixing packaging tests in `intelmq-api`? Would be nice to have them working

Time is extremely limited, I'm unsure if it's possible in the next weeks.
